packages/generate-newsletter/index.js

```diff
@@ -0,0 +1,98 @@
+const fs = require('fs-extra')
+const path = require('path')
+const htmlCreator = require('html-creator')
+const { prompt } = require('enquirer')
+const { extractSheets } = require('spreadsheet-to-json')
+const argv = require('yargs').argv
+
+const { getRandomColor, colorLuminance, getCurrentDate } = require('./utils')
+const credentials = require('./credentials.json')
+
+require('dotenv').load() // Load .env and put it into process.env
+
+const spreadsheetKey = process.env.SHEET_URL
+const outDir = argv.out || '.'
+
+const askQuestions = async () => {
+  const userResponse = await prompt([
+    {
+      type: 'input',
+      name: 'issue',
+      message: 'Issue no ?',
+    },
+    {
+      type: 'autocomplete',
+      name: 'month',
+      message: `Select the newsletter's month`,
+      suggest(input, choices) {
+        return choices.filter(choice => choice.message.startsWith(input))
+      },
+      choices: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
+    },
+    {
+      type: 'input',
+      name: 'date',
+      message: 'Date ?',
+      initial: getCurrentDate(),
+    },
+  ])
+
+  return userResponse
+}
+
+askQuestions().then(userResponse => {
+  console.log('\n Generating pls wait... \n ')
+
+  extractSheets({ spreadsheetKey, credentials }, (err, data) => {
+    if (err) {
+      console.log('ERROR: ', err)
+      return
+    }
+    let linksCollection = Object.values(data)[0].map(item => {
+      return `
+				<tr>
+					<td>
+						<div class="issue__content">
+							<a href="${item.LINK}"><h3 class="issue__content-title">${item.TITLE}</h3></a>
+							<p class="issue__content-desc">${item.DESCRIPTION}</p>
+							<div class="issue__content-info"><a href="${item.LINK}">${item.LINK}</a> <span>${item.AUTHOR}</span></div>
+						</div>
+					</td>
+				</tr>`
+    })
+    let links = linksCollection.join('')
+    const html = `
+<center>
+	<table align="center" border="0" cellspacing="0" width="100%" height="100%" cellpadding="0">
+	<tbody>${links}</tbody>
+	</table>
+</center>	
+		`
+    const currentYear = userResponse.date.split('-').shift()
+    const month = userResponse.month.substring(0, 3).toLowerCase()
+    const issueNo = userResponse.issue
+    // @TODO: update the content title
+    const markdown = `---
+path: "/issues/${currentYear}/${month}/${issueNo}"
+published: true
```

Comment:
can we make it as false. We should manually enable this when we are publishing newsletter every time. It sort of a check for us to verify everything before pushing.

Reply:
👍